### PR TITLE
Cyclesort in JavaScript

### PR DESCRIPTION
**Pull request by ireneelizabethsabu**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
YES
(Write your answer here.)

### Description
Adding cycleSort algorithm in JavaScript.
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#2505 
(Write your answer here.)

### CODE REVIEW

File: JavaScript/sort/cycleSort.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+input = [1, 8, 7, 61, 5, 4, 11];
+EXPECTED: [1,4,5,7,8,11,61]
+
+CASE 2:
+input = [1,8,3,9,10,10,2,4];
+EXPECTED: [1,2,3,4,8,9,10,10]
+
+    Time Complexity - O(n^2)
+    Space Complexity - O(1)
+*/
```
Comment (HarshCasper) [inline JavaScript/sort/cycleSort.js]: Add a new line at the EOF

File: JavaScript/sort/cycleSort.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
+/*CYCLE SORT:
+Cycle sort is an in-place unstable sorting Algorithm that is optimal in terms of the 
+total number of writes to the original array.
+*/
```
Comment (HarshCasper) [inline JavaScript/sort/cycleSort.js]: Add a more descriptive pitch

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by ireneelizabethsabu** [comment]
@HarshCasper please review the changes made.

File: JavaScript/sort/cycleSort.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
+/*CYCLE SORT:
+Cycle sort is an in-place unstable sorting Algorithm that is optimal in terms of the total number of writes to the original array.
+Consider a list with n elements. For an element item, we find the index at which it should occur in the sorted list by counting the smaller elements in a. If the element is already at the correct position, the iteration is skipped, otherwise
+we will write it to its intended position. If that position is inhabited by a different element , it should also be moved to its correct position. This process must be continued until an element is moved to the original position. This completes a cycle. Such n cycles are executed.
+*/
```
Comment (HarshCasper) [inline JavaScript/sort/cycleSort.js]: Modularize this Content

File: JavaScript/sort/cycleSort.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+CASE 2:
+input = [1,8,3,9,10,10,2,4];
+EXPECTED: [1,2,3,4,8,9,10,10]
+
+    Time Complexity - O(n^2)
+    Space Complexity - O(1)
+
+*/
+
```
Comment (HarshCasper) [inline JavaScript/sort/cycleSort.js]: Remove the extra line

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Rest is good to merge :+1:

**Comment by ireneelizabethsabu** [comment]
@HarshCasper Updated cycleSort.js. Please look into it.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**